tests/fuzz.py

```diff
@@ -249,32 +261,45 @@ def fuzz(tests_dir, fuzz_arg, num_iter):
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description='Fuzzing KECC.')
-    parser.add_argument('-n', '--num', type=int, help='The number of tests')
+    parser.add_argument('-n', '--num', type=int, help='The number of tests', default=None)
     parser.add_argument('-p', '--print', action='store_true', help='Fuzzing C AST printer')
     parser.add_argument('-i', '--irgen', action='store_true', help='Fuzzing irgen')
     parser.add_argument('-r', '--reduce', action='store_true', help="Reducing input file")
+    parser.add_argument('--skip-build', action='store_true', help="Skipping cargo build")
+    parser.add_argument('--easy', action='store_true', help="Generate more easy code by csmith option")
+    parser.add_argument('--seed', type=int, help="Provide seed of fuzz generation", default=-1)
     args = parser.parse_args()
 
     if args.print and args.irgen:
         raise Exception("Choose an option used for fuzzing: '--print' or '--irgen', NOT both")
+    
     if args.print:
         fuzz_arg = "-p"
     elif args.irgen:
         fuzz_arg = "-i"
     else:
         raise Exception("Specify fuzzing argument")
 
+    if args.seed != -1:
+        print('Set seed as', args.seed)
+        random.seed(args.seed)
+    else:
+        print('Use default random seed')
+
     tests_dir = os.path.abspath(os.path.dirname(__file__))
 
-    print("Building KECC..")
-    try:
-        proc = subprocess.Popen(["cargo", "build", "--release"], cwd=tests_dir)
-        proc.communicate()
-    except subprocess.TimeoutExpired as e:
-        proc.kill()
-        raise e
+    if not args.skip_build:
+        print("Building KECC..")
+        try:
+            proc = subprocess.Popen(["cargo", "build", "--release"], cwd=tests_dir)
+            proc.communicate()
+        except subprocess.TimeoutExpired as e:
+            proc.kill()
+            raise e
+    else: 
+        print("Skip building. You should manually build the binary. Please execute `cargo build --release` to build.")
```

Comment:
I'm not sure this option will be useful because `cargo` already skip the build if there was no change in the source code. (https://doc.rust-lang.org/cargo/reference/build-scripts.html#life-cycle-of-a-build-script)

What do you think? @gmlwns2000 @jeehoonkang

Reply:
I agree with @minseongg

Reply:
I thought `cargo run` does not build the binary, so I could skip the `cargo build` and use the pre-built binary to save time.

It was my misunderstanding, and I checked `cargo run` actually update and build the binary before running.

So I agree this change is useless.

Reply:
@gmlwns2000 Now I understand the purpose of `--skip-build` option.

I added a commit to your branch which add `--skip-build`. ([`552d560`](https://github.com/kaist-cp/kecc-public/pull/2/commits/552d560525c9eee491d10b36d0029151ab7323f2)) Please check it works in your implementation!

Reply:
@minseongg I checked your commit, and it works well wonderful. 

By skipping the build process, the whole fuzz process is way faster now. Thank you for committing it.